datarepo-clienttests/src/main/java/testscripts/BuildSnapshotWithFiles.java

```diff
@@ -0,0 +1,123 @@
+
+    // wait for the job to complete
+    bulkLoadArrayJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, bulkLoadArrayJobResponse);
+    BulkLoadArrayResultModel result =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, bulkLoadArrayJobResponse, BulkLoadArrayResultModel.class);
+    BulkLoadResultModel loadSummary = result.getLoadSummary();
+    assertThat(
+        "Number of successful files loaded should equal total files.",
+        loadSummary.getTotalFiles(),
+        equalTo(loadSummary.getSucceededFiles()));
+
+    // generate load for the simple datase
+    String testConfigGetIngestbucket = "jade-testdata";
+    String fileRefName =
+        "scratch/buildSnapshotWithFiles/" + FileUtils.randomizeName("input") + ".json";
+    IngestRequestModel ingestRequest =
+        BulkLoadUtils.makeIngestRequestFromLoadArray(
+            result, testConfigGetIngestbucket, fileRefName);
+
+    // load the data
+    JobModel ingestTabularDataJobResponse =
+        repositoryApi.ingestDataset(datasetSummaryModel.getId(), ingestRequest);
+
+    ingestTabularDataJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, ingestTabularDataJobResponse);
+    IngestResponseModel ingestResponse =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestTabularDataJobResponse, IngestResponseModel.class);
+    logger.info("Successfully loaded data into dataset: {}", ingestResponse.getDataset());
+
+    for (Integer batchSize : batchSizes) {
+      logger.info("Setting batch size to {}", batchSize);
+      DataRepoUtils.setConfigParameter(repositoryApi, "FIRESTORE_SNAPSHOT_BATCH_SIZE", batchSize);
+
+      // create the snapshots with all of the files
+      List<SnapshotSummaryModel> snapshotSummaryModels = new ArrayList<>();
+      for (int i = 0; i < snapshotsToCreate; i++) {
+        JobModel createSnapshotJobResponse =
+            DataRepoUtils.createSnapshot(
+                repositoryApi, datasetSummaryModel, "snapshot-simple.json", true);
+
+        SnapshotSummaryModel snapshotSummaryModel =
+            DataRepoUtils.expectJobSuccess(
+                repositoryApi, createSnapshotJobResponse, SnapshotSummaryModel.class);
+        logger.info("Successfully created snapshot: {}", snapshotSummaryModel.getName());
+        snapshotSummaryModels.add(snapshotSummaryModel);
+      }
+
+      for (SnapshotSummaryModel snapshotSummaryModel : snapshotSummaryModels) {
+        JobModel deleteSnapshotJobResponse =
+            repositoryApi.deleteSnapshot(snapshotSummaryModel.getId());
+        deleteSnapshotJobResponse =
+            DataRepoUtils.waitForJobToFinish(repositoryApi, deleteSnapshotJobResponse);
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, deleteSnapshotJobResponse, DeleteResponseModel.class);
+        logger.info("Successfully deleted snapshot: {}", snapshotSummaryModel.getName());
+      }
```

Comment:
It looks like this calls the simple dataset cleanup method.  In addition, should we handle the case that the test fails before the snapshots are cleaned up?

Reply:
Good suggestion. I changed it to collect the created snaps and delete them during cleanup.